nikola/plugins/task/indexes.py

```diff
@@ -27,320 +27,74 @@
-                                                     extension=extension)
-
-    def index_section_path(self, name, lang, is_feed=False, is_rss=False):
-        """Link to the index for a section.
-
-        Example:
-
-        link://section_index/cars => /cars/index.html
-        """
-        extension = None
-
-        if is_feed:
-            extension = ".atom"
-            index_file = os.path.splitext(self.site.config['INDEX_FILE'])[0] + extension
-        elif is_rss:
-            index_file = 'rss.xml'
-        else:
-            index_file = self.site.config['INDEX_FILE']
-        if name in self.number_of_pages_section[lang]:
-            number_of_pages = self.number_of_pages_section[lang][name]
-        else:
-            posts = [post for post in self._get_filtered_posts(lang, self.site.config['SHOW_UNTRANSLATED_POSTS']) if post.section_slug(lang) == name]
-            number_of_pages = self._compute_number_of_pages(posts, self.site.config['INDEX_DISPLAY_POST_COUNT'])
-            self.number_of_pages_section[lang][name] = number_of_pages
-        return utils.adjust_name_for_index_path_list([_f for _f in [self.site.config['TRANSLATIONS'][lang],
-                                                                    name,
-                                                                    index_file] if _f],
-                                                     None,
-                                                     utils.get_displayed_page_number(None, number_of_pages, self.site),
-                                                     lang,
-                                                     self.site,
-                                                     extension=extension)
-
-    def index_atom_path(self, name, lang):
-        """Link to a numbered Atom index.
-
-        Example:
-
-        link://index_atom/3 => /index-3.atom
-        """
-        return self.index_path(name, lang, is_feed=True)
-
-    def index_section_atom_path(self, name, lang):
-        """Link to the Atom index for a section.
-
-        Example:
-
-        link://section_index_atom/cars => /cars/index.atom
-        """
-        return self.index_section_path(name, lang, is_feed=True)
-
-    def index_section_rss_path(self, name, lang):
-        """Link to the RSS feed for a section.
-
-        Example:
-
-        link://section_index_rss/cars => /cars/rss.xml
-        """
-        return self.index_section_path(name, lang, is_rss=True)
+        kw.update(context)
+        return context, kw
+
+    def postprocess_posts_per_classification(self, posts_per_classification_per_language, flat_hierarchy_per_lang=None, hierarchy_lookup_per_lang=None):
+        """Rearrange, modify or otherwise use the list of posts per classification and per language."""
+        for lang, posts_per_classification in posts_per_classification_per_language.items():
+            if len(posts_per_classification) == 0:
+                posts_per_classification[""] = []
```

Comment:
I think I wrote that before adding support for `get_implicit_classifications`, which handles this more generally. (It was also used in that plugin.) I've removed this piece.